tests/integration/security/tls_config/tls_config_test.go

```diff
@@ -0,0 +1,135 @@
+)
+
+func TestTlsVersion13Config(t *testing.T) {
+	framework.NewTest(t).
+		Features("security.tls.config").
+		Run(func(ctx framework.TestContext) {
+			if ctx.Clusters().IsMulticluster() {
+				t.Skip("https://github.com/istio/istio/issues/37307")
+			}
+			instA := apps.A[0]
+			instB := apps.B[0]
+			// First check TLS certificate
+			certPEMs := cert.DumpCertFromSidecar(t, instA, instB, "http")
+			block, _ := pem.Decode([]byte(strings.Join(certPEMs, "\n")))
+			if block == nil { // nolint: staticcheck
+				t.Fatalf("failed to parse certificate PEM")
+			}
+			// Check B responds OK for the call from A on mTLS
+			opts := echo.CallOptions{
+				To: instB,
+				Port: echo.Port{
+					Name: "http",
+				},
+				Check: check.And(
+					check.OK(),
+					check.MTLSForHTTP()),
+				Retry: echo.Retry{
+					Options: []retry.Option{retry.Delay(10 * time.Second), retry.Timeout(30 * time.Second)},
+				},
+			}
+			instA.CallOrFail(t, opts)
+
+			target := fmt.Sprintf("%s.%s:%d", util.BSvc, apps.Namespace1.Name(), 80)
+			fromSelector := fmt.Sprintf("app=%s", util.ASvc)
+			kubeConfig := (ctx.Clusters().Default().(*kube.Cluster)).Filename()
+			fromPod, err := dir.GetPodName(apps.Namespace1, fromSelector, kubeConfig)
+			if err != nil {
+				t.Fatalf("err getting the pod from pod name: %v", err)
+			}
+			retry := tutil.Retrier{
+				BaseDelay: 10 * time.Second,
+				Retries:   3,
+				MaxDelay:  30 * time.Second,
+			}
+
+			// Check that TLS-1.1 is disallowed
+			err = checkTLSVersion(false, "-tls1_1", kubeConfig, retry, apps.Namespace1, fromPod, "istio-proxy", target)
+			if err != nil {
+				t.Errorf("Checking TLS 1.1 disallowed returns error: %v", err)
+			} else {
+				t.Logf("as expected, TLS 1.1 is disallowed")
+			}
+			// Check that TLS-1.2 is disallowed
+			err = checkTLSVersion(false, "-tls1_2", kubeConfig, retry, apps.Namespace1, fromPod, "istio-proxy", target)
+			if err != nil {
+				t.Errorf("Checking TLS 1.2 disallowed returns error: %v", err)
+			} else {
+				t.Logf("as expected, TLS 1.2 is disallowed")
+			}
+			// Check that TLS-1.3 is allowed
+			err = checkTLSVersion(true, "-tls1_3", kubeConfig, retry, apps.Namespace1, fromPod, "istio-proxy", target)
+			if err != nil {
+				t.Errorf("Checking TLS 1.3 allowed returns error: %v", err)
+			} else {
+				t.Logf("as expected, TLS 1.3 is allowed")
+			}
+		})
+}
+
+// Two cases:
+// 1. Check a TLS version is allowed when allow is true.
+// 2. Check a TLS version is disallowed when allow is false.
+func checkTLSVersion(allow bool, tlsVer, kubeConfig string, retry tutil.Retrier, ns namespace.Instance, fromPod, fromContainer, connectTarget string) error {
+	checkString := "Cipher is (NONE)"
+	var expResult string
+	if allow {
+		expResult = "allowed"
+	} else {
+		expResult = "disallowed"
+	}
+
+	retryFn := func(_ context.Context, i int) error {
+		execCmd := fmt.Sprintf(
+			"kubectl exec %s -c %s -n %s --kubeconfig %s -- openssl s_client -alpn istio %s -connect %s",
```

Comment:
Depending on `openssl` in the container is not safe since we support open source. if you want to get the TLS certifiates from the handshake, the echo client can do this with the `TLS` protocol.

Reply:
The openssl is used here to verify whether a specific TLS version is allowed/disallowed during handshake through the -tls1_1, -tls1_2, and -tls1_3 flags of openssl s_client command. It is not used to get TLS certificates. AFAIK, the echo client does not support verifying a specific TLS version is allowed/disallowed during handshake.

Reply:
Based on the comments, the integration test has been removed.